src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -85,7 +93,9 @@ public class ChangeStreamOperation<TResult> : IChangeStreamOperation<TResult>
         private readonly IReadOnlyList<BsonDocument> _pipeline;
         private ReadConcern _readConcern = ReadConcern.Default;
         private readonly IBsonSerializer<TResult> _resultSerializer;
+
         private BsonDocument _resumeAfter;
+        private BsonDocument _startAfter;
         private BsonTimestamp _startAtOperationTime;
```

Comment:
Add a few new fields:

    private BsonDocument _documentResumeToken;
    private BsonTimestamp _initialOperationTime;

The first tracks the most recently seen document resume token.

The second is used (if required) to track the operation time of the initial aggregate command.

Reply:
Done